From 6056405454c0cee670d59adff7858f635b018c29 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Wed, 14 Dec 2016 11:38:49 -0600
Subject: [PATCH 1349/5242] MLK-13612-1: ARM: dts: enable QSPI for i.MX7ULP1

commit  15ce2158ba2d11658746d1db46e6e8efc4f3383e from
https://source.codeaurora.org/external/imx/linux-imx.git

Add one more dts for QSPI on i.MX7ULP1 for mfgtool purpose.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile             |    1 +
 arch/arm/boot/dts/imx7ulp-evk-qspi.dts |   41 ++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx7ulp.dtsi         |    3 +++
 3 files changed, 45 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7ulp-evk-qspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b56bfc5..0e05297 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -697,6 +697,7 @@ dtb-$(CONFIG_SOC_IMX7ULP) += \
 	imx7ulp-evk.dtb \
 	imx7ulp-evk-emmc.dtb \
 	imx7ulp-evk-lpuart.dtb \
+	imx7ulp-evk-qspi.dtb \
 	imx7ulp-evk-wm8960.dtb
 dtb-$(CONFIG_SOC_LS1021A) += \
 	ls1021a-moxa-uc-8410a.dtb \
diff --git a/arch/arm/boot/dts/imx7ulp-evk-qspi.dts b/arch/arm/boot/dts/imx7ulp-evk-qspi.dts
new file mode 100644
index 0000000..93dbb07
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-evk-qspi.dts
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7ulp-evk.dts"
+
+&qspi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_qspi1_1>;
+	status = "okay";
+
+	flash0: mx25r6435f@0 {
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25r6435f";
+		spi-max-frequency = <29000000>;
+	};
+};
+
+&iomuxc1 {
+	imx7ulp-evk {
+		pinctrl_qspi1_1: qspi1grp_1 {
+			fsl,pins = <
+				IMX7ULP_PAD_PTB7_LLWU0_P11__QSPIA_SS1_B    0x43 /* SS1 */
+				IMX7ULP_PAD_PTB8__QSPIA_SS0_B              0x43 /* SS0 */
+				IMX7ULP_PAD_PTB15__QSPIA_SCLK              0x43 /* SCLK */
+				IMX7ULP_PAD_PTB9_LLWU0_P12__QSPIA_DQS      0x43 /* DQS */
+				IMX7ULP_PAD_PTB16_LLWU0_P14__QSPIA_DATA3   0x43 /* D3 */
+				IMX7ULP_PAD_PTB17__QSPIA_DATA2             0x43 /* D2 */
+				IMX7ULP_PAD_PTB18__QSPIA_DATA1             0x43 /* D1 */
+				IMX7ULP_PAD_PTB19_LLWU0_P15__QSPIA_DATA0   0x43 /* D0 */
+			>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 67a2216..71b8bee 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -532,6 +532,9 @@
 			reg = <0x410A5000 0x10000>, <0xC0000000 0x10000000>;
 			reg-names = "QuadSPI", "QuadSPI-memory";
 			interrupts = <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clks IMX7ULP_CLK_DUMMY>,
+				 <&clks IMX7ULP_CLK_DUMMY>;
+			clock-names = "qspi_en", "qspi";
 			status = "disabled";
 		};
 
-- 
1.7.9.5

